site/docs/develop/spec.md

```diff
@@ -1,80 +1,180 @@
 # Nessie Specification
 
-This page documents the complete nessie specification. This includes:
+This page documents the complete Nessie specification. This includes:
 
 * API and its constraints
 * Contract for value objects
-* requirements for backend storage implementations
 
 ## API contract
 
-!!! warning
-    todo
+The Nessie API is used by Nessie integrations within for example Apache Iceberg or Delta Lake and
+user facing applications like Web UIs.
 
-## Contract for Value Objects
+Nessie defines a [REST API](rest.md) (OpenAPI) and reference implementations for [Java](java.md)
+and [Python](python.md).
+
+## Contents in Nessie
 
 ### General Contract
 
+_Content Objects_ describe the state of a data lake object like a table or view.
+Nessie currently provides types for Iceberg tables, Delta Lake tables and SQL views. Nessie uses
+two identifiers for a single Content object:
+
+1. The [Content Id](#content-id) is used to identify a content object across all branches even
+   if the content object is been referred to using different table or view names.
+2. The [Content Key](#content-key) is used to look up a content object by name, like a table name
+   or view name. The Content Key changes when the referred table or view is renamed.
+
+#### Content Key
+
+The Content Key consists of multiple strings and is used to resolve a symbolic name, like a table
+name or a view name used in SQL statements, to a Content object.
+
+When a table or view is renamed using for example an SQL `ALTER TABLE RENAME` operation, Nessie
+will record this operation using a _remove_ operation plus a _put_ operation
+([see below](#operations)).
```

Comment:
should be `#operations-in-a-nessie-commit ?

Reply:
Good catch! Fixed